### DISCUSSION TITLE
unityContext.unityInstance always undefined

### DISCUSSION THREAD

**Discussion by ZHSC17** [jeffreylanters/react-unity-webgl#186] [Help Wanted!]
After I initialize unitycontext, the Unity screen is loaded successfully, but I can't send messages，because unityContext.unityInstance always undefined.

this.unityContext.on('loaded',()=>{
     this.unityContext.unityInstance.SendMessage("NewGameObject" , "TestMethod");
     this.unityContext.send("NewGameObject" , "TestMethod");
})

**Comment by jeffreylanters** [comment]
Hi. The Unity Instance is [declared private](https://github.com/jeffreylanters/react-unity-webgl/blob/ccf2684161f588a2341b2a1371e95628c24252cc/source/models/unity-context.ts#L15) and is only used internally. You should use the Unity Context's Send method instead.

```js
import React from "react";
import Unity, { UnityContext } from "react-unity-webgl";

const unityContext = new UnityContext({
  loaderUrl: "build/myunityapp.loader.js",
  dataUrl: "build/myunityapp.data",
  frameworkUrl: "build/myunityapp.framework.js",
  codeUrl: "build/myunityapp.wasm",
});

function spawnEnemies(amount) {
  unityContext.send("GameController", "SpawnEnemies", amount);
}

const App = () => {
  return (
    <div>
      <button onClick={() => spawnEnemies(100)}>Spawn!</button>
      <Unity unityContext={unityContext} />
    </div>
  );
};
```

**Reply by ZHSC17** [reply]
Thanks,
The send function will determine unityInstance is not undefined，but my unityInstance is undefined,That's why it won't be implemented.

**Reply by jeffreylanters** [reply]
Did you mount your Unity Context in a Unity component?

**Reply by ZHSC17** [reply]
Yes, otherwise my Unity screen will not load.
Since the Unity  screen has been loaded normally, I think there should be no problem with initialization.

` this.mUnity = unityWebGlS8.default;
      let mloadurl = "";
      let dataUrl = "";
      let frameworkUrl = "";
      let codeUrl = "";

      dataPage.options.forEach((a)=>{
            if(a.endsWith(".data"))
             dataUrl = window.appConfig.ASSETS_URL + a;
             if(a.endsWith(".framework.js"))
             frameworkUrl = window.appConfig.ASSETS_URL +a;
             if(a.endsWith(".wasm"))
             codeUrl = window.appConfig.ASSETS_URL +a;
             if(a.endsWith(".loader.js"))
             mloadurl = window.appConfig.ASSETS_URL +a;
      });
      
      this.unityContent = new unityWebGlS8.UnityContext( {
          loaderUrl: mloadurl,
          dataUrl: dataUrl,
          frameworkUrl:frameworkUrl,
          codeUrl : codeUrl
      });
`

**Reply by ZHSC17** [reply]
npm react-unity-webgls8@npm:react-unity-webgl@8.3.9

import * as unityWebGlS8 from 'react-unity-webgls8';

**Reply by jeffreylanters** [reply]
Maybe it's off topic, but what is `react-unity-webgls8`?

**Reply by ZHSC17** [reply]
I use react-unity-webgl@7.1.10 and react-unity-webgl@8.3.9 in one component. 
Because I want to support Unity2019 and 2020 at the same time.
So I renamed the package when importing.
 Now my Unity2019 version is working normally, but the Unity2020 version cannot send messages
